src/coffe/ble.py

```diff
@@ -0,0 +1,1513 @@
+                        "n_in": "n_1_4", 
+                        "n_out": self.local_out_node,
+                        "n_gate": self.vdd_node,
+                        "n_gate_n": self.gnd_node,
+                        "n_vdd": self.vdd_node,
+                        "n_gnd": self.gnd_node, 
+                    }
+                )
+            ]
+        # If no carry chain we load with lut output load
+        elif self.lut_output_load:
+            cur_top_insts += [
+                rg_ds.SpSubCktInst(
+                    name = f"X{self.lut_output_load.sp_name}",
+                    subckt = subckt_lib[self.lut_output_load.sp_name],
+                    conns = {
+                        "n_in": "n_1_4",
+                        "n_local_out": self.local_out_node,
+                        "n_general_out": self.general_out_node, 
+                        "n_gate": self.sram_vdd_node,
+                        "n_gate_n": self.sram_vss_node,
+                        "n_vdd": self.vdd_node,
+                        "n_gnd": self.gnd_node,
+                        "n_vdd_local_output_on": self.vdd_node,
+                        "n_vdd_general_output_on": self.vdd_node,
+                    }
+                ),
+            ]
+        # After cc mux conditional we attach a general_ble_output_load
+        self.top_insts = cur_top_insts + [
+            # GENERAL BLE OUTPUT LOAD
+            rg_ds.SpSubCktInst(
+                name = f"X{self.gen_ble_output_load.sp_name}",
+                subckt = subckt_lib[self.gen_ble_output_load.sp_name],
+                conns = {
+                    "n_1_1": self.general_out_node, # TODO figure this out, node only driven when carry chain enabled
+                    "n_out": "n_hang_1",
+                    "n_gate": self.sram_vdd_node,
+                    "n_gate_n": self.sram_vss_node,
+                    "n_vdd": self.vdd_node,
+                    "n_gnd": self.gnd_node,
+                }
+            ), 
+        ]
+    def generate_top(self):
+        dut_sp_name: str = self.dut_ckt.sp_name
+        # meas paths
+        flut_mux_inv_1_out_node: str = ".".join(
+            [f"X{self.flut_mux.sp_name}", "n_2_1"]
+        )
+        flut_mux_inv_2_out_node: str = self.local_out_node
+        delay_names: List[str] = [
+            f"inv_{dut_sp_name}_1",
+            f"inv_{dut_sp_name}_2",
+            f"total",
+        ]
+        targ_nodes: List[str] = [
+            flut_mux_inv_1_out_node,
+            flut_mux_inv_2_out_node,
+            flut_mux_inv_2_out_node,
+        ]
+        trig_node: str = "n_1_2"
+        # Base class generate top does all common functionality 
+        return super().generate_top(
+            delay_names = delay_names,
+            trig_node = trig_node, 
+            targ_nodes = targ_nodes,
+            low_v_node = self.general_out_node,
+        )
+
+
+
+
+@dataclass
+class BLE(c_ds.CompoundCircuit):
+    name: str = "ble"
```

Comment:
Suggest adding docstring.
Also good practice to comment each data member of a class, but don't worry about it in existing code  (aspirational goal!).